luminance-sdl2/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+//! [SDL2](https://crates.io/crates/sdl2) backend for [luminance](https://crates.io/crates/luminance)
+//! and [luminance-windowing](https://crates.io/crates/luminance-windowing).
+
+#![deny(missing_docs)]
+
+use gl;
+use luminance::context::GraphicsContext;
+use luminance::framebuffer::Framebuffer;
+use luminance::framebuffer::FramebufferError;
+use luminance::texture::Dim2;
+pub use luminance_gl::gl33::StateQueryError;
+use luminance_gl::GL33;
+pub use luminance_windowing::{CursorMode, WindowDim, WindowOpt};
+pub use sdl2;
+use std::fmt;
+use std::os::raw::c_void;
+
+/// Error that can be risen while creating a surface.
+#[derive(Debug)]
+pub enum Sdl2SurfaceError {
+  /// Initialization of the surface went wrong.
+  InitError(String),
+  /// Window creation failed.
+  WindowCreationFailed(sdl2::video::WindowBuildError),
+  /// Failed to create an OpenGL context.
+  GlContextInitFailed(String),
+  /// No available video mode.
+  VideoInitError(String),
+  /// The graphics state is not available.
+  ///
+  /// This error is generated when the initialization code is called on a thread on which the
+  /// graphics state has already been acquired.
+  GraphicsStateError(StateQueryError),
+}
+
+impl fmt::Display for Sdl2SurfaceError {
+  fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+    match *self {
+      Sdl2SurfaceError::InitError(ref e) => {
+        f.write_str("initialization error: ")?;
+        e.fmt(f)
```

Comment:
You can simplify with

```suggestion
        write!(f, "initialization error: {}", e)
```

Reply:
Yea, I don't know what I was thinking when I decided to do this microoptimization. It's probably not even faster, and even if it was, it's not worth the uglier code. I'll change it.